src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -128,31 +137,36 @@ private string ResolveClassName(AssociativeNode identifierList)
                 return String.Empty;
 
             var resolvedName = elementResolver.LookupResolvedName(partialName);
-            if (string.IsNullOrEmpty(resolvedName))
-            {
-                // If namespace resolution map does not contain entry for partial name, 
-                // back up on compiler to resolve the namespace from partial name
-                var matchingClasses = CoreUtils.GetResolvedClassName(classTable, identifierList);
+            if (!string.IsNullOrEmpty(resolvedName)) 
+                return resolvedName;
+            
+            // If namespace resolution map does not contain entry for partial name, 
+            // back up on compiler to resolve the namespace from partial name
+            var matchingClasses = CoreUtils.GetResolvedClassName(classTable, identifierList);
 
-                if (matchingClasses.Length == 1)
-                {
-                    resolvedName = matchingClasses[0];
-                    var assemblyName = CoreUtils.GetAssemblyFromClassName(classTable, resolvedName);
+            if (matchingClasses.Length == 1)
+            {
+                resolvedName = matchingClasses[0];
+                var assemblyName = CoreUtils.GetAssemblyFromClassName(classTable, resolvedName);
 
-                    elementResolver.AddToResolutionMap(partialName, resolvedName, assemblyName);
-                }
+                elementResolver.AddToResolutionMap(partialName, resolvedName, assemblyName);
+            }
+            else if (matchingClasses.Length > 1)
+            {
+                OnLogSymbolConflictWarning(partialName, matchingClasses);
```

Comment:
`resolvedName` is an empty string for this case, shouldn't it be the first element in `matchingClasses`?